File: backend/src/main/java/com/back/domain/book/category/entity/Category.java
package com.back.domain.book.category.entity;

import com.back.domain.book.book.entity.Book;
import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor
public class Category extends BaseEntity {
    String name;

    @OneToMany(mappedBy = "category")
    private List<Book> books = new ArrayList<>();

    public Category(String name) {
        this.name = name;
    }
}

File: backend/src/main/java/com/back/domain/book/category/entity/Category.kt
package com.back.domain.book.category.entity

import com.back.domain.book.book.entity.Book
import com.back.global.jpa.entity.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.OneToMany

@Entity
class Category(
    var name: String
) : BaseEntity() {

    @OneToMany(mappedBy = "category")
    private val _books: MutableList<Book> = mutableListOf()

    val books: List<Book>
        get() = _books.toList()

    fun addBook(book: Book) {
        _books.add(book)
    }

    fun removeBook(book: Book) {
        _books.remove(book)
    }
}